### ISSUE TITLE
Add seed to output directory name

### ISSUE THREAD

**Issue by mfeurer** [automl/SMAC3#209]
In order to have an overview of what has been run one would sometimes like to know the seed of previous runs. One solution would be to store it in the output directory name, but this isn't as easy as it sounds as the scenario object does not know the seed for the SMAC object.

**Comment by KEggensperger** [comment]
A remark: We don't use a seed but a rng.

Nevertheless, for the CLI there exists a seed and adding that to the folder name still makes sense.

**Comment by KEggensperger** [comment]
Sorry, but I think this is not yet fully fixed:

```from smac.facade.smac_facade import SMAC
from smac.scenario.scenario import Scenario
from smac.configspace import ConfigurationSpace
from ConfigSpace.hyperparameters import UniformIntegerHyperparameter
import numpy as np

def dummy(cfg, seed):
    return 1

cs = ConfigurationSpace()
a = UniformIntegerHyperparameter("a", 0, 10, default=5)
cs.add_hyperparameter(a)

scenario = Scenario({
        "run_obj": "quality",
        "runcount-limit": 100,
        "cs": cs,
        "output_dir": "./a"})

smac = SMAC(scenario=scenario, tae_runner=dummy, rng=2)
smac = SMAC(scenario=scenario, tae_runner=dummy, rng=np.random.RandomState(2))
```

I would expect that at least one of these cmds writes to a directory "a/run2", but both write to "a_run1". Apparently, the seed/rng does not affect the name of the ```output_dir```

**Comment by mlindauer** [comment]
no, you have to pass the `run_id` to `Scenario`. The `scenario` object is right now responsible for creating the output directory and in `SMAC` it is already to late to use the `rng`. The alternative would be that the optimizer (here `SMAC`) has to create the `output_directory`; however this would imply that every optimizer is responsible for the same task.

btw: why have you closed the issue again?

**Comment by KEggensperger** [comment]
I clicked the wrong button ..

Yes, that's what @mfeurer already pointed out. What about adding the seed to the Scenario instead of the optimizer?

**Comment by mlindauer** [comment]
Our seed handling is already a bit strange (in some classes). I would say that seeds should be passed on a function-level and not on a class-level. I don't want to recreate all objects again only because I want to run the same stuff with a different seed.
The same applies for the scenario object. To run SMAC again with a different seed, I would not like to recreate the scenario.

**Comment by mfeurer** [comment]
@mlindauer @KEggensperger is this fixed by #240 or #334 ?

**Comment by mlindauer** [comment]
more or less solved by #334. 
We save now the run_id (or the seed on the cli, resp.) in the output dir name, i.e., the run_id passed to the scenario object. 
However, if a Python user does not specify a run_id, the output dor will always include run_id 1 regardless of the rng object passed to the facade.

**Comment by mlindauer** [comment]
@mfeurer do we have a solution for this issue now?

**Comment by mfeurer** [comment]
Not really - should we sample a random number from the RNG if no `run_id` is provided via the python interface?

**Comment by mlindauer** [comment]
Ok.

**Comment by shukon** [comment]
Fixed with #418
